### PR TITLE
Issue #14937: Migrated section 5.2.8 type variable names to chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java [hunk 1]
```diff
@@ -10,49 +10,51 @@
  * pattern = "(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"
  *
  */
-public record InputRecordTypeParameterName<t>(Integer x, String str) { // warn
+public record InputRecordTypeParameterName<t>(Integer x, String str) {
+    // violation above 'Record type name 't' must match pattern'
     public <TT> void foo() { }
 
-    <e_e> void foo(int i) {
+    <T> void foo(int i) {
```
Comment (romani) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java:17]: please share a reason on such updates.
Weird code is good for testing.

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java [hunk 2]
```diff
@@ -10,49 +10,51 @@
...
+// violation below 'Record type name 'foo' must match pattern'
+record Other <foo extends Serializable & Cloneable>
 (LinkedHashMap<String, Node> linkedHashMap) {
 
     foo getOne() {
         return null;//comment
     }
 
-    <Tfo$o2T extends foo> /*comment*/Tfo$o2T getTwo(Tfo$o2T a) {
+    <T extends foo> /*comment*/T getTwo(T a) {
```
Comment (romani) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java:29]: same as above, looks like all is changed in this way

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java (hunk 2 above)
Comment (Zopsss) [reply src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java:29]: I have replied you above, pls check that out

**Comment by Zopsss** [comment]
rebased the pr

**Comment by Zopsss** [comment]
Circle CIs failing, dont know why

**Review by romani — APPROVED** [review]
Ok to merge.
Original Inputs were created as copy from test Inputs, in ITs we can relax a bit and do less crazy input, as we need to focus on most used by user and realistic examples

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
@Zopsss please resolve conflicts

**Pull request merged**